### ISSUE TITLE
Custom Fieldtype - {options:my_field_name} not available in Channel Form

### ISSUE THREAD

**Issue by rowansentesy** [ExpressionEngine/ExpressionEngine#746]
This is an issue dating back to EE3 I believe, and I am curious about why this functionality was disabled. 

In system/ee/ExpressionEngine/Addons/channel/libraries/channel_form/Channel_form_lib.php line 2834, there is reference to a $this->custom_option_fields array. This section is commented out in EE3 - EE6. 

We are updating one of our addons for EE6, and my idea was to enable the {option:FIELD_NAME} loop for Channel Form to allow more customization and styling of the form inputs. 

I would prefer to use the core ee functionality similar to 'select', but this appears to be impossible. 

I can get it to work if I add my fieldname to this array: (Line 2771)
`        $this->native_option_fields = array(
            'multi_select',
            'select',
            'radio',
            'checkboxes', 'my_fieldtype_name'
        );
`

Obviously this is not going to work for an addon, so I am looking for another way to do this. 

I would like to be able to register a 'custom_option_field' through my addon code. 

Can someone advise me and let me know if this is possible? If it isn't, and there is a good reason, maybe you can illuminate as to why this chunk of code (line 2834) was abandonned? 

Thanks so much!

Rowan Sentesy
Hop Studios

**Comment by intoeetive** [comment]
@rowansentesy are you building a fieldtype? Based on what I see in the docs, [`{options:my_field_name}`](https://docs.expressionengine.com/latest/channels/channel-form/overview.html#optionsmy_field_name) should be available if you have your fieldtype declare itself with `extends OptionFieldtype` (similar to what `ft.checkboxes.php` does) - did you by chance try that?

As for the code that's commented out my guess is that it was removed because the extension hooks referenced SafeCracker, but never got added back - @robinsowell do you have any more knowledge about that?

**Comment by rowansentesy** [comment]
Hi @intoeetive,
Thanks for the response to this. I think that is the exact tip I was looking for. I will try the `extends OptionFieldtype` suggestion.
Yes, we are updating a fieldtype & addon to be EE6 compatible and looking at ways to make it better and more compliant with recent EE core work.

Thanks again, I will keep you posted.
Rowan

**Comment by intoeetive** [comment]
Sounds good!

I'll close this issue - please reopen or open new one if you experience any issues

**Comment by rowansentesy** [comment]
Hey @intoeetive,

I am carrying on with this and used the `extends OptionFieldType` as you suggested, referencing the `ft.checkboxes.php` and that seems to be OK, although not providing me with what I'm needing here. 

I'm still having trouble with how to register the fieldtype in the `public $custom_option_fields` array (Channel_form_lib.php line 28). 

I can't find any examples of where this array is written to. 
Can you show me how to register our fieldtype in this array? Even if it is registered, I don't think it will show up unless the Channel_form_lib.php is modified... 

`ft.checkboxes.php` doesn't provide me with any help on how to access this array for our fieldtype. Checkboxes are defined in the `native_option_fields` array. 

I'm trying to call the 'get_field_options' but its not returning anything because our fieldtype is not in the 'options_fields' array. 
See: Channel_form_lib.php line 2444

Basically, I'm still stuck. 

If it is not possible, then I will pivot, I just think this is so close to working, I'd rather use these core features.

Thanks,
Rowan

**Comment by intoeetive** [comment]
@rowansentesy do I understand the problem correctly that the options for the 'OptionFieldType' are not being populated in Channel Form? 
I think the `option_fields` needs to be built by checking which of actually installed fieldtypes extend `OptionFieldType` - however also need to make sure it does not have any side implications. That being said, we probably can't provide any solution before 6.0.1 release - unless you would be willing to fork off https://github.com/ExpressionEngine/ExpressionEngine/tree/6.x and submit pull request.

I'm re-opening the ticket and classifying as feature request.

**Comment by rowansentesy** [comment]
Hey @intoeetive 
Thanks for the info & update. I have discussed it with the Hop Studios team and we are interested in forking & submitting a pull request. I will be in touch if we have any questions. 
@HopStudios-Development  @gilbert-hop

Thanks,
Rowan

**Comment by gilbert-hop** [comment]
@intoeetive I created a pull request that enables the custom option fieldtype declaration. Please have a look :)
